src/System.Net.Mail/src/System/Net/Mail/SmtpTransport.cs

```diff
@@ -205,9 +206,14 @@ internal void EndGetConnection(IAsyncResult result)
                 throw new ArgumentNullException(nameof(recipients));
             }
 
-            if (GlobalLog.IsEnabled && recipients.Count > 0)
+            if (recipients.Count <= 0)
```

Comment:
I assume this was a bug?  It's now asserting in the opposite condition.  Regardless, this seems like a strange assert.  Presumably _recipients is provided by the end user?  Are we validating that it's not empty prior to this call?  If this is coming from user input, we should not be asserting.